### PR TITLE
Feature/lc-42: populate sidebar with chat sessions

### PR DESCRIPTION
**Pull request by jcm5155**

### CODE REVIEW

File: frontend/labs-chat/src/app/api/chatSessions/route.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { dynamoDBDocumentClient } from "@helpers/aws";
+import { QueryCommand } from "@aws-sdk/lib-dynamodb";
+import { cookies } from "next/headers";
+import { runWithAmplifyServerContext } from "@helpers/amplify-server-utils";
+import { getCurrentUser } from "aws-amplify/auth/server";
+
+export async function GET(): Promise<Response> {
+  const scanLimit = 20;
```
Comment (jcm5155) [inline frontend/labs-chat/src/app/api/chatSessions/route.ts:8]: I'm not a huge fan of the behavior of filtering + scanLimit. The filter always happens AFTER the scanLimit, so this means that any `is_hidden` sessions returned will still count against the total number of sessions returned (resulting in some number of documents LESS than the max if any are hidden). We'll have to think about our design here if we want to remedy this, which seemed out of scope for this story.

File: frontend/labs-chat/src/app/components/ChatHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import ChatHistoryListItem from "@components/ChatHistoryListItem";
+import { getFormattedDateForUI } from "@helpers/dates";
+import { Session } from "../types";
```
Comment (scaperoth) [inline frontend/labs-chat/src/app/components/ChatHistory.tsx:11]: should this be @/types or something?

File: frontend/labs-chat/src/helpers/dates.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Returns the given Date in a formatted string.
+ * The format of the string is [YYYY-MM-DD] HH:MM.
+ *
+ * @param {Date} dt The date to format.
+ * @returns {string} The formatted date and time string.
+ */
+export const getFormattedDateForUI = (dt: Date): string => {
```
Comment (superjkm) [inline frontend/labs-chat/src/helpers/dates.ts:8]: this is really sensible

File: frontend/labs-chat/src/app/components/ChatHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import ChatHistoryListItem from "@components/ChatHistoryListItem";
+import { getFormattedDateForUI } from "@helpers/dates";
+import { Session } from "@/app/types";
```
Comment (superjkm) [inline frontend/labs-chat/src/app/components/ChatHistory.tsx:11]: is this @app/types?

**Review by NAPTlME — APPROVED** [review]

**Pull request merged**